### PR TITLE
Return -FI_ENOSYS from fi_av_open if a named (shared) AV is requested

### LINKED ISSUE #976: Disallow named AVs (shared AVs)

### PR DESCRIPTION
**Pull request by sungeunchoi**
@chuckfossen @ztiffany 

Fixes #976 

Signed-off-by: Sung-Eun Choi sungeunchoi@users.noreply.github.com

### CODE REVIEW

**Comment by ztiffany** [comment]
:+1:

File: prov/gni/src/gnix_av.c [hunk 1]
```diff
@@ -82,6 +82,13 @@ static int gnix_verify_av_attr(struct fi_av_attr *attr)
 		break;
 	}
 
+	if (attr->name != NULL) {
+		ret = -FI_ENOSYS;
+	}
+
+	GNIX_WARN(FI_LOG_AV,
+		  "%d %p %d\n", attr->type, attr->name, ret);
+
```
Comment (chuckfossen) [inline prov/gni/src/gnix_av.c]: This looks like it could be GNIX_INFO

**Review by chuckfossen — APPROVED** [review]
With caveat

File: prov/gni/src/gnix_av.c (hunk 1 above)
Comment (sungeunchoi) [reply prov/gni/src/gnix_av.c]: Crap -- that's debug output I forgot about.  I'll remove.

**Comment by sungeunchoi** [comment]
@chuckfossen Updated

**Comment by chuckfossen** [comment]
👍

**Pull request merged**